### DISCUSSION TITLE
Micromamba-based VS Code devcontainer

### DISCUSSION THREAD

**Discussion by maresb** [mamba-org/micromamba-docker#176] [Show and tell]
https://github.com/maresb/micromamba-devcontainer

I made this devcontainer image based on Micromamba-Docker. It makes it easy and efficient to bootstrap a Conda environment without having a predetermined Python version. It also sets up Docker-from-Docker, and includes many small helpful features like being preconfigured in case one installs [pre-commit](https://pre-commit.com/) or uses a `.git-blame-ignore-revs` file.

It required a lot of tinkering to get Git and Docker-from-Docker running to my satisfaction, and I've found it invaluable for managing the development environments for myself and my collaborators. It's a work-in-progress, but I think it's now mature enough to share, and I'm very interested to hear any feedback, critical or otherwise. I hope some people find it useful!

**Comment by wholtz** [comment]
Looking good! 

On line 21 you copy the `docker-compose` binary  from the `docker-compose` image. Then on line 127 you overwrite the previously copied file. Is that just some old cruft you haven't cleaned up or is there some logic behind that process?

**Reply by maresb** [reply]
Oops, good catch! Thanks for spotting that.

(And even if there were some logic, I would have been missing an explanatory comment.)

Still very "work-in-progress" so it's great to get some fresh eyes. Thank you!
